exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/StringFunctions.java

```diff
@@ -529,7 +538,16 @@ public void eval() {
 
```

Comment:
Isn't this too complicated ?   I tried and changed this code by adding here:
```
      String orig = new String(buffer.unsafeGetMemory(0,out.end));
      String upp = orig.toUpperCase(/*Locale.GERMAN*/);
      buffer.setBytes(0,upp.getBytes(),0,out.end);
```
and then commented out the whole following *for* loop, and the test passed!

(not sure which one performs better)

Reply:
I don't know reason (maybe performance or allocation of big strings ?), but In test classes you can find simple implemenations:
org.apache.drill.udf.dynamic.LowerFunction 
/ org.apache.drill.udf.dynamic.CustomUpperFunction

Reply:
These functions are generated to test dynamic UDFs, they are not shipped with Drill.